src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (!victim.hidden && victim instanceof Herbivore) {
+      Animal.alive.forEach(animal => {
+        animal.health -= 50;
+      });
```

Comment:
In the https://github.com/mate-academy/js_herbivores_and_carnivores/pull/1237#discussion_r1020319204 Marina meant that you don't need a loop at all (now you reduce health for all animals if only one has been bitten). You already have an animal that should lose health - it is `victim`.